docs/book/src/jobs.md

```diff
@@ -0,0 +1,42 @@
+# Jobs
+
+This document intents to provide an overview over our jobs running via Prow and Github actions.
+
+## Build and Tests running on the default branch
+
+
+ > NOTE : To see which test jobs execute which e2e tests, you can click on the links which lead to the
+  respective test overviews in testgrid.
+
+### Presubmits
+
+  Prow Presubmits:
+
+  - [pull-cluster-api-provider-azure-build] `./scripts/ci-build.sh` REQUIRED
+  - [pull-cluster-api-provider-azure-test] `./scripts/ci-test.sh`   REQUIRED
+  - [pull-cluster-api-provider-azure-e2e] `./scripts/ci-e2e.sh`     REQUIRED
+  - [pull-cluster-api-provider-azure-e2e-exp] `./scripts/ci-e2e.sh`
+  - [pull-cluster-api-provider-azure-windows] `./scripts/ci-e2e.sh` REQUIRED
+  - [pull-cluster-api-provider-azure-verify]                        REQUIRED
+  - [pull-cluster-api-provider-azure-apidiff] `./scripts/ci-apidiff.sh`
```

Comment:
checks that there are no API changes and fails if there are, the point is to remind reviewers to check that breaking changes are handled in conversion, the job is not marked as required for merge